packages/api/src/schema/schema.yaml

```diff
@@ -811,6 +813,354 @@ components:
             created
           example: 1587667174725
 
+    vod-asset:
+      type: object
+      table: vod_asset
+      required:
+        - name
+      properties:
+        id:
+          type: string
+          readOnly: true
+          example: 09F8B46C-61A0-4254-9875-F71F4C605BC7
+          index: true
+        userId:
+          type: string
+          example: 66E2161C-7670-4D05-B71D-DA2D6979556F
+          index: true
+          description: owner of the asset
+        deleted:
+          type: boolean
+          description: Set to true when the asset is deleted
+        objectStoreId:
+          type: string
+          description: Object store ID where the asset is stored
+          example: 09F8B46C-61A0-4254-9875-F71F4C605BC7
+        name:
+          type: string
+          description:
+            Name of the asset. This is not necessarily the filename, can be a
+            custom name or title
+          example: filename.mp4
+        duration:
+          type: number
+          description: Duration of the asset in seconds (float)
+          example: 23.8328
+        aspectRatio:
+          type: string
+          description: Aspect ratio of the asset
+          example: 16:9
+        meta:
+          type: object
+          description: User input metadata associated with the asset
+          example:
+            {
+              "title": "My awesome video",
+              "description": "This is a video of my awesome life",
+              "tags": ["awesome", "life", "video"],
+            }
+        updatedAt:
+          type: string
+          description:
+            Timestamp (in milliseconds) at which the asset was last updated
+          example: 2022-01-01T02:20:00Z
+        createdAt:
+          type: string
+          description: Timestamp (in milliseconds) at which asset was created
+          example: 2022-01-01T02:20:00Z
+        format:
```

Comment:
This might fit in the `videoParams` or `videoSpec` or something like that.

WDYT of also making this an`enum` with only the formats that we support?

Reply:
I think the `format` should stay on the top level of the asset, as this is pretty specific to the container and other fields depend on the container of the asset.

About the enum sounds great, we actually could make separate types for the formats, as they come with different logic / parameters sometimes, wdyt?

Reply:
But aren't containers a video-specific thing? So they could be inside the params/spec about `video`-kind of assets, but not necessarily every kind of asset should have a `format`. Even if they do tho, like for example an `image` one might still have a format `jpeg`, the enum values will end up being different anyway, so it's not really the same field, it only has the same name. 

What I am thinking overall is to have the same pattern that we discussed for `Task`s input/output parameters, so that there is both a top-level "type" field which specifies what is the type of the asset, and then there is a specific field with the name of that `type`, with the specific parameters that are only relevant for that specific type. And that can be infinitely nested. 

So for example could have a `type: video` then `video.format: mp4` and then additional `mp4` fields under `video.mp4: {}`, like this:
```
id: 123
name: People Are Awesome
objectStoreId: 345
size: 3476384
type: video
video:
  format: mp4
  width: ...
  height: ...
  bitrate: ...
  mp4:
    anyCustomMp4Stuff: ...
```

That way, on the root of the `asset` object, we only hold directly the fields that would be valid for almost any kind of "file" in general. Then we specify what type of file it is, then have a dedicated sub-object for that type, etc.

On our MVP we will only have `video` files. But soon we might already have `hls` as a separate `type` altogether (or just as a separate `format`? Maybe not since it's not even a single file). And later we might want to have other things like `image` perhaps, for thumbnails or even GIFs or something.

Reply:
Sounds good, we could put all these fields and the ones you cited in the other comments under a video field. It is for sure clearer, if we plan on using the same Asset for other type of assets (images, NFT, audios..)  
At this point I would rename from `vod-asset` to `asset`, as vod is video specific, wdyt?
I also wonder if in a second step we should separate all the metadata in a separate place in the schema

Reply:
Just asset instead of vod-asset makes sense to me as well! And we can separate the metadata, but I'm ok with keeping it on the root now as, after all, this whole object is metadata anyway. So we got the common file metadata on the root and the specialized metadata in the category-specific fields.